lib/versioner/trashcan.go

```diff
@@ -164,10 +157,13 @@ func (t *Trashcan) cleanoutArchive() error {
 		return err
 	}
 
-	// The last directory seen by the walkFn may not have been removed as it
-	// should be.
-	if currentDir != "" && filesInDir == 0 {
-		t.fs.Remove(currentDir)
+	for _, path := range filesCounter.emptyDirs() {
```

Comment:
While your at it: This entire for loop could also be a convenience function like `func (c *filesCounter) deleteEmptyDirs(fs fs.filesystem)`.

Reply:
Done. I decided not to return an error from `deleteEmptyDirs` and instead left a log warning if you don't mind.